File: CS/SchedulerGSync/SchedulerGSync/Helpers/ApplicationConfig.cs
using System;
using System.IO;
using System.Linq;
using System.Xml.Serialization;

namespace SchedulerGSync {
    public class ApplicationConfig {
        public static ApplicationConfig Instance { get; private set; } = Load();

        static ApplicationConfig Load() {
            var fileInfo = new FileInfo($"{nameof(SchedulerGSync)}.appsettings");
            if (!fileInfo.Exists)
                return new ApplicationConfig();
            XmlSerializer serializer = new XmlSerializer(typeof(ApplicationConfig));
            using (var fileStream = fileInfo.OpenRead())
                return (ApplicationConfig)serializer.Deserialize(fileStream);
        }

        public ApplicationConfig() { }

        public string CalendarId { get; set; }

        public void Save() {
            var fileInfo = new FileInfo($"{nameof(SchedulerGSync)}.appsettings");
            XmlSerializer serializer = new XmlSerializer(typeof(ApplicationConfig));
            using (var fileStream = fileInfo.OpenWrite())
                serializer.Serialize(fileStream, this);
        }
    }
}

File: VB/SchedulerGSync/SchedulerGSync/Helpers/ApplicationConfig.vb
Imports System
Imports System.IO
Imports System.Linq
Imports System.Xml.Serialization

Namespace SchedulerGSync
	Public Class ApplicationConfig
		Private Shared privateInstance As ApplicationConfig = Load()
		Public Shared Property Instance() As ApplicationConfig
			Get
				Return privateInstance
			End Get
			Private Set(ByVal value As ApplicationConfig)
				privateInstance = value
			End Set
		End Property

		Shared Function Load() As ApplicationConfig
			Dim fileInfo = New FileInfo($"{NameOf(SchedulerGSync)}.appsettings")
			If Not fileInfo.Exists Then
				Return New ApplicationConfig()
			End If
			Dim serializer As New XmlSerializer(GetType(ApplicationConfig))
			Using fileStream = fileInfo.OpenRead()
				Return DirectCast(serializer.Deserialize(fileStream), ApplicationConfig)
			End Using
		End Function

		Public Sub New()
		End Sub

		Public Property CalendarId() As String

		Public Sub Save()
			Dim fileInfo = New FileInfo($"{NameOf(SchedulerGSync)}.appsettings")
			Dim serializer As New XmlSerializer(GetType(ApplicationConfig))
			Using fileStream = fileInfo.OpenWrite()
				serializer.Serialize(fileStream, Me)
			End Using
		End Sub
	End Class
End Namespace
